### PR TITLE
Hunger thirst systems

### PR DESCRIPTION
**Pull request by Slarvsylt**
System for thirst and hunger. When thirst/hunger reaches 1.0f a DeathEvent-component is attached to the animal. Also added a hybrid class for getting and decreasing hunger/thirst-levels.

### CODE REVIEW

File: Assets/Scripts/ECS/Animal/Needs/ThirstSystem.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                thirstData.Thirst += Time.DeltaTime / 1000.0f;
+
+                if(thirstData.Thirst > 1.0f)
+                {
+                    commandBuffer.AddComponent<DeathEvent>(entityInQueryIndex, entity);
+                }
+
+            }).ScheduleParallel();
+
+            throw new NotImplementedException();
```
Comment (gimme) [inline Assets/Scripts/ECS/Animal/Needs/ThirstSystem.cs]:
```suggestion
            m_EndSimulationEcbSystem.AddJobHandleForProducer(Dependency);
```

File: Assets/Scripts/ECS/Animal/Needs/HungerSystem.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            {
+                hungerData.Hunger += Time.DeltaTime / 1000.0f;
+
+                if(hungerData.Hunger <= 0.0f)
+                {
+                    commandBuffer.AddComponent<DeathEvent>(entityInQueryIndex, entity);
+                }
+
+            }).ScheduleParallel();
+            throw new NotImplementedException();
```
Comment (gimme) [inline Assets/Scripts/ECS/Animal/Needs/HungerSystem.cs]:
```suggestion

            m_EndSimulationEcbSystem.AddJobHandleForProducer(Dependency);
```

File: Assets/Scripts/ECS/Animal/Needs/HungerSystem.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        protected override void OnUpdate()
+        {
+            var commandBuffer = m_EndSimulationEcbSystem.CreateCommandBuffer().ToConcurrent();
+
+            Entities.ForEach((Entity entity, int entityInQueryIndex,
+                ref HungerData hungerData) =>
+            {
+                hungerData.Hunger += Time.DeltaTime / 1000.0f;
+
+                if(hungerData.Hunger <= 0.0f)
```
Comment (gimme) [inline Assets/Scripts/ECS/Animal/Needs/HungerSystem.cs:35]:
```suggestion
                if(hungerData.Hunger > 1.0f)
```

File: Assets/Scripts/ECS/Hybrid/NeedsStatus.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            if (add)
+            {
+                entityManager.AddComponentData(entity, component);
+            }
+            else
+            {
+                entityManager.RemoveComponent<T>(entity);
+            }
+        }
+
```
Comment (gimme) [inline Assets/Scripts/ECS/Hybrid/NeedsStatus.cs]:
```suggestion
```

File: Assets/Scripts/ECS/Hybrid/NeedsStatus.cs [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        /// <summary>
+        /// Decrease the thirst of an animal when drinking.
+        /// </summary>
+        /// <param name="value">Float value between 0 and 1</param>
+        public void DecreaseThirst(float value)
+        {
+            float cur = GetComp<ThirstData>().Thirst;
+            entityManager.SetComponentData<ThirstData>(entity, new ThirstData { Thirst = cur - value });
+        }
```
Comment (gimme) [inline Assets/Scripts/ECS/Hybrid/NeedsStatus.cs]: The decrease methods can reduce the values below 0. Is this intended?

**Review by gimme — CHANGES_REQUESTED** [review]

**Review by gimme — COMMENTED** [review]
I think that hunger/thirst would be easier to understand as like a "hunger bar" that **decreases** as you get **more** hungry.

I also think that it would be better if it was a number from 0 to infinity instead of 0 to 1 as that would work well with different max hungers for different animals.

**Comment by Slarvsylt** [comment]
I comply with your requests.

**Review by gimme — APPROVED** [review]

**Pull request merged**